### DISCUSSION TITLE
Hi : What are the versions need to migrate from php7.3 to php8.0?

### DISCUSSION THREAD

**Discussion by JainulMufafika** [rectorphp/rector#7901] [Q&A] [answered]
Hi,
I am trying to migrate some PHP projects from 7.3 to 8.1 version with Rector, I have some doubts.

1. When we migrate from php7.3 to php8.0 , if we need to migrate to only php8.0 or UPTO 8.0.
UPTO 8.0  set method(which apply upto 8.0)
 $rectorConfig->sets([
        LevelSetList::UP_TO_PHP_80
    ]);

2.what are the rules have to be applied for entire migrate process
In the below link 420 rules are mentioned, if we need to apply all the rules which is in the link for migrating php7.3 to php 8.0?
https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#arguments
what are the rules have to be applied for migration process from php7.3 to php 8.0

3. For migrating from php7.3 to php 8.0 . First we need to migrate php7.3 to php7.4 and then migrating php7.4 to php8.0. Or we can directly migrate php8.0 directly

**Comment by samsonasik** [comment, marked as answer]
On very beginning on legacy project with many files, I suggest to use per-Set list SetList::PHP_XX first for easy to review the change, then continue until no error.

When all applied, you can use the up to `LevelSetList::UP_TO_PHP_80` so the code will be compatible for new code,
